lib/grist-api.ts

```diff
@@ -75,7 +80,7 @@ export class GristDocAPI {
     this._server = options.server || 'https://api.getgrist.com';
     this._apiKey = options.apiKey || null;
     this._chunkSize = options.chunkSize || 500;
-    const match = /^(https?:.*)\/doc\/([^\/#]+)/.exec(docUrlOrId);
+    const match = /^(https?:.*)\/doc\/([^\/?#]+)/.exec(docUrlOrId);
```

Comment:
This pattern won't match new-style links. If the first /part/ of the url path is longer than some threshold (looks to be 12 characters), it can be assumed to be a urlId.

Reply:
I updated. It makes for an uglier regexp, but seems to work.